pkgs/stdenv/darwin/default.nix

```diff
@@ -5,15 +5,15 @@
   if localSystem.isAarch64 then
     let
       fetch = { file, sha256, executable ? true }: import <nix/fetchurl.nix> {
-        url = null; # to be built by hydra and injected here
+        url = "http://tarballs.nixos.org/stdenv-darwin/aarch64/20acd4c4f14040485f40e55c0a76c186aa8ca4f3/${file}";
         inherit (localSystem) system;
         inherit sha256 executable;
       }; in {
-        sh      = fetch { file = "sh";    sha256 = "0000000000000000000000000000000000000000000000000000"; };
-        bzip2   = fetch { file = "bzip2"; sha256 = "0000000000000000000000000000000000000000000000000000"; };
-        mkdir   = fetch { file = "mkdir"; sha256 = "0000000000000000000000000000000000000000000000000000"; };
-        cpio    = fetch { file = "cpio";  sha256 = "0000000000000000000000000000000000000000000000000000"; };
-        tarball = fetch { file = "bootstrap-tools.cpio.bz2"; sha256 = "0000000000000000000000000000000000000000000000000000"; executable = false; };
+        sh      = fetch { file = "sh";    sha256 = "17m3xrlbl99j3vm7rzz3ghb47094dyddrbvs2a6jalczvmx7spnj"; };
```

Comment:
My hash doesnt' match?

```
[grahamc@hyperchicken:~]$ nix hash to-base32 --type sha256 b4bb70d824de3c7c95d3d187479d092fda9fcec1a963a1a38ea6d0dcc978588c
132qg34xrl56isis2qx9q779znig16flg1yisfapqg6y4kc71fxl
```

Reply:
It's explained in the OP.  Executables don't use plain hashes.

Reply:
The reason is that the plain file hash does not express the executable bit in any way, so plain-hashed paths in /nix/store are always without it.